packages/contractkit/src/utils/phone-number-lookup/phone-number-lookup.ts

```diff
@@ -97,27 +98,39 @@ export async function postToPhoneNumPrivacyService<ResponseType>(
   }
 
   const { pgpnpUrl } = context
-  const res = await fetch(pgpnpUrl + endpoint, {
-    method: 'POST',
-    headers: {
-      Accept: 'application/json',
-      'Content-Type': 'application/json',
-      Authorization: authHeader,
-    },
-    body: bodyString,
-  })
-
-  if (!res.ok) {
+
+  const tries = 3
+  let lastError
+  for (let i = 1; i <= tries; i++) {
+    const res = await fetch(pgpnpUrl + endpoint, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        Authorization: authHeader,
+      },
+      body: bodyString,
+    })
+
+    if (res.ok) {
+      debug('Response ok. Parsing.')
+      const response = await res.json()
+      return response as ResponseType
+    }
+
     debug(`Response not okay. Status ${res.status}`)
+
     switch (res.status) {
       case 403:
```

Comment:
Would appreciate feedback on this part. Are there any other error codes we want to immediately throw after (and not retry) ?

Reply:
Today, we also return `400` (user input error), `401` (unauthenticated), and `500` (unknown server error). I think `400` and `401` we should not retry, since they're guaranteed to fail.